Frameworks/CoreText/DWriteWrapper_CTFont.mm

```diff
@@ -208,6 +217,203 @@ HRESULT _DWriteCreateFontFaceWithFontDescriptor(CTFontDescriptorRef fontDescript
+        ComPtr<IDWriteFontFamily> family;
+        RETURN_IF_FAILED(font->GetFontFamily(&family));
+        ComPtr<IDWriteLocalizedStrings> dwriteFamilyName;
+        RETURN_IF_FAILED(family->GetFamilyNames(&dwriteFamilyName));
+        CFStringRef fontFamilyName = _CFStringFromLocalizedString(dwriteFamilyName.Get());
+        if (__CFStringEqual(familyName, fontFamilyName)) {
+            matchingFonts.emplace_back(font);
+        }
+    }
+
+    return S_OK;
+}
+
+static HRESULT __FilterFontsByInformationalString(const std::vector<ComPtr<IDWriteFont>>& fonts,
+                                                  CFStringRef name,
+                                                  DWRITE_INFORMATIONAL_STRING_ID id,
+                                                  std::vector<ComPtr<IDWriteFont>>& matchingFonts) {
+    for (auto& font : fonts) {
+        BOOL exists;
+        ComPtr<IDWriteLocalizedStrings> dwriteString;
+        RETURN_IF_FAILED(font->GetInformationalStrings(id, &dwriteString, &exists));
+        if (exists) {
+            CFStringRef fontName = _CFStringFromLocalizedString(dwriteString.Get());
+            if (__CFStringEqual(name, fontName)) {
+                matchingFonts.emplace_back(font);
+            }
+        }
+    }
+
+    return S_OK;
+}
+
+static HRESULT __FilterFontsByKeyAndValue(std::vector<ComPtr<IDWriteFont>>& fonts, CFStringRef key, CFTypeRef value, bool forceMatch) {
+    std::vector<ComPtr<IDWriteFont>> matchingFonts;
+    matchingFonts.reserve(fonts.size());
+    CFRange range{ 0, CFStringGetLength(key) };
+    if (__CFStringEqual(key, kCTFontFamilyNameAttribute)) {
+        RETURN_IF_FAILED(__FilterFontsByFamilyName(fonts, static_cast<CFStringRef>(value), matchingFonts));
+    } else if (__CFStringEqual(key, kCTFontNameAttribute)) {
+        RETURN_IF_FAILED(__FilterFontsByInformationalString(fonts,
+                                                            static_cast<CFStringRef>(value),
+                                                            DWRITE_INFORMATIONAL_STRING_POSTSCRIPT_NAME,
+                                                            matchingFonts));
+    } else if (__CFStringEqual(key, kCTFontDisplayNameAttribute)) {
+        RETURN_IF_FAILED(__FilterFontsByInformationalString(fonts,
+                                                            static_cast<CFStringRef>(value),
+                                                            DWRITE_INFORMATIONAL_STRING_FULL_NAME,
+                                                            matchingFonts));
+    } else if (__CFStringEqual(key, kCTFontStyleNameAttribute)) {
+        RETURN_IF_FAILED(__FilterFontsByStyleName(fonts, static_cast<CFStringRef>(value), matchingFonts));
+    } else if (__CFStringEqual(key, kCTFontTraitsAttribute)) {
+        __FilterFontsByTraits(fonts, static_cast<CFDictionaryRef>(value), matchingFonts);
+    } else {
+        // Other attributes are currently unsupported by CTFont
+        UNIMPLEMENTED();
```

Comment:
`UNIMPLEMENTED_WITH_MSG` may be more helpful, since these keys are strings and their values can be logged.